File: go/wam/6692.go
const WamStatusViewerAction = 6692 // channel: regular

type StatusViewerActionEvent struct {
	AttributionType TOPBARATTRIBUTIONTYPE `wam:"attribution_type"` // field 1
	ViewerActionType STATUSVIEWACTIONTYPE `wam:"viewer_action_type"` // field 2
	AttributionTypes string `wam:"attribution_types"` // field 3
	StatusCategory STATUSCATEGORY `wam:"status_category"` // field 4
	ExternalSourceDomainType INLINEVIDEOTYPE `wam:"external_source_domain_type"` // field 5
	UrlStatusClicked URLSTATUSCLICKED `wam:"url_status_clicked"` // field 6
	UrlStatusType URLSTATUSTYPE `wam:"url_status_type"` // field 7
	ActionIndex int64 `wam:"action_index"` // field 8
	LinkLoadTime int64 `wam:"link_load_time"` // field 9
	LinkOpenResult string `wam:"link_open_result"` // field 10
	PsaCampaignId string `wam:"psa_campaign_id"` // field 11
	StatusId string `wam:"status_id"` // field 12
	StatusViewerSessionId int64 `wam:"status_viewer_session_id"` // field 13
}

File: ts/wam/6692.ts
export const WamStatusViewerAction = 6692 as const

export interface StatusViewerActionEvent {
  /** field 1, wire `attribution_type` */
  attributionType?: TOPBARATTRIBUTIONTYPE
  /** field 2, wire `viewer_action_type` */
  viewerActionType?: STATUSVIEWACTIONTYPE
  /** field 3, wire `attribution_types` */
  attributionTypes?: string
  /** field 4, wire `status_category` */
  statusCategory?: STATUSCATEGORY
  /** field 5, wire `external_source_domain_type` */
  externalSourceDomainType?: INLINEVIDEOTYPE
  /** field 6, wire `url_status_clicked` */
  urlStatusClicked?: URLSTATUSCLICKED
  /** field 7, wire `url_status_type` */
  urlStatusType?: URLSTATUSTYPE
  /** field 8, wire `action_index` */
  actionIndex?: number
  /** field 9, wire `link_load_time` */
  linkLoadTime?: number
  /** field 10, wire `link_open_result` */
  linkOpenResult?: string
  /** field 11, wire `psa_campaign_id` */
  psaCampaignId?: string
  /** field 12, wire `status_id` */
  statusId?: string
  /** field 13, wire `status_viewer_session_id` */
  statusViewerSessionId?: number
}